File: 2109. Adding Spaces to a String.js
/**
 * @param {string} s
 * @param {number[]} spaces
 * @return {string}
 */
var addSpaces = function(s, spaces) {
    let result = [];
    let prev = 0;

    for(let index of spaces){
        result.push(s.slice(prev,index));
        result.push(" ");
        prev = index;
    }


    result.push(s.slice(prev));
    return result.join("");
};

File: 2109. Adding Spaces to a String.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        result = []
        prev=0

        for index in spaces:
            result.append(s[prev:index])
            result.append(" ")
            prev = index
        
        result.append(s[prev:])
        return "".join(result)